auth/auth.go

```diff
@@ -1,21 +1,50 @@
 package auth
 
 import (
+	"io/ioutil"
+	"net/http"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/golang/glog"
 	"github.com/rancherio/host-api/app/common"
 	"github.com/rancherio/host-api/config"
-	jwt "github.com/dgrijalva/jwt-go"
-	"net/http"
 )
 
+var fileBytes []byte
+
 func Auth(rw http.ResponseWriter, req *http.Request) bool {
 	if !config.Config.Auth {
+		glog.Infoln("Host_API auth is disabled")
 		return true
 	}
+	getToken := req.URL.Query().Get("token")
 
-	token, err := jwt.ParseFromRequest(req, func(token *jwt.Token) (interface{}, error) {
-		return config.Config.Key, nil
+	if len(getToken) == 0 {
+		return false
+	}
+	glog.Infoln("Enabling Authencation for Host_API")
+
+	if len(fileBytes) == 0 {
+		file, err := ioutil.ReadFile(config.Config.Key)
+		if err != nil {
+			glog.Error("Error reading file")
+			return false
+		}
+		fileBytes = file
+	} else {
+		glog.Infoln("Reading cached content")
+	}
+	token, err := jwt.Parse(getToken, func(token *jwt.Token) (interface{}, error) {
+		return fileBytes, nil
 	})
 
+	if err == nil && token.Valid {
```

Comment:
Yeah i'll improve the log message there. I wanted to imply that a valid token has been found, but authentication can still be unsuccessful at that point since we have an additional check on the hostUuid.